src/shogun/features/StringFeatures.cpp

```diff
@@ -78,19 +78,22 @@ template<class ST> CStringFeatures<ST>::CStringFeatures(CAlphabet* alpha)
 }
 
 template<class ST> CStringFeatures<ST>::CStringFeatures(const CStringFeatures & orig)
-: CFeatures(orig), num_vectors(orig.num_vectors),
-	single_string(orig.single_string),
-	length_of_single_string(orig.length_of_single_string),
-	max_string_length(orig.max_string_length),
-	num_symbols(orig.num_symbols),
-	original_num_symbols(orig.original_num_symbols),
-	order(orig.order), preprocess_on_get(false),
-	feature_cache(NULL)
+: CFeatures(orig)
 {
 	init();
 
 	ASSERT(orig.single_string == NULL) //not implemented
 
+	num_vectors = orig.num_vectors;
+	single_string = orig.single_string;
+	length_of_single_string = orig.length_of_single_string;
+	max_string_length = orig.max_string_length;
+	num_symbols = orig.num_symbols;
+	original_num_symbols = orig.original_num_symbols;
+	order = orig.order;
+	preprocess_on_get = false;
+	feature_cache = NULL;
+
```

Comment:
Isn't it better practice to do list initialisation, i.e. how it was before? Since this is a copy constructor all the class members should haven been initialised anyway

Reply:
The init() is called in the constructor which resets all the values to default. The unit test we have now doesn't catch his problem.

Reply:
And I am not sure if this is entirely correct because single_string is a pointer. How we have it now creates a shallow copy which I am not sure if it's what we want.

Reply:
>The init() is called in the constructor which resets all the values to default. The unit test we have now doesn't catch his problem.

Ah ok! Sorry I didn't see that. 

>And I am not sure if this is entirely correct because single_string is a pointer. How we have it now creates a shallow copy which I am not sure if it's what we want.

True, that doesn't seem like a good idea. I am guessing if you delete the original CStringFeatures the copy has undefined behaviour?

Reply:
Yes, that's right. I will make another PR for that after the merge of this one.